mlflow/models/docker_utils.py

```diff
@@ -37,6 +37,20 @@
 RUN pip install virtualenv
 """
 
+if os.getenv("http_proxy") is not None and os.getenv("https_proxy") is not None:
+    MAVEN_PROXY = (
+        " -DproxySet=true -Dhttp.proxyHost={http_proxy_host} "
+        "-Dhttp.proxyPort={http_proxy_port} -Dhttps.proxyHost={https_proxy_host} "
+        "-Dhttps.proxyPort={https_proxy_port} -Dhttps.nonProxyHosts=repo.maven.apache.org"
+    ).format(
+        http_proxy_host=os.environ["http_proxy"].split("//")[1].split(":")[0],
+        http_proxy_port=os.environ["http_proxy"].rsplit(":", maxsplit=1)[1],
```

Comment:
why not use `http_proxy_port=os.environ["http_proxy"].split("//")[1].split(":")[1]` ? any difference ?

Reply:
@WeichenXu123 because `http_proxy_port=os.environ["http_proxy"].rsplit(":", maxsplit=1)[1]` is simpler.

Reply:
@WeichenXu123 Curious why you think we should use `http_proxy_port=os.environ["http_proxy"].split("//")[1].split(":")[1]`?

Reply:
@harupy and @WeichenXu123 thanks for your comments, my intention for using the split approach was to forcefully fail the code, if the proxy is not adhering to the standard format `http://HOST:PORT`, but I'll update the code (and add comments) to make use of `urllib.parse.urlparse' and update the PR.